### DISCUSSION TITLE
Include a mention to "sideEffects" in the internal packages documentation

### DISCUSSION THREAD

**Discussion by vimtor** [vercel/turborepo#6363] [Ideas]
To provide context, I was transitioning from a single Next.js project to a monorepo. As part of this transition, I created an internal package for my Prisma database and updated the imports from `"../lib/prisma"` to `"@acme/prisma"`. However, this change resulted in a new error that started appearing on some Next.js pages:

```
PrismaClient is unable to be run in the browser.
```

This was unexpected because I hadn't made any other changes. Upon investigation, I realized that the error was caused by importing types into client components and the Prisma client being bundled in the client code:

```ts
import { User } from "@prisma/client";
```

For the record, I should probably import the above using ﻿`import type`, but that's beyond the point. Additionally, I needed to import some enum values as values, not types.

I recalled that for tree-shaking to work, [the package should be side-effect free](https://webpack.js.org/guides/tree-shaking/#:~:text=The%20new%20webpack%204%20release%20expands%20on%20this%20capability%20with%20a%20way%20to%20provide%20hints%20to%20the%20compiler%20via%20the%20%22sideEffects%22%20package.json%20property%20to%20denote%20which%20files%20in%20your%20project%20are%20%22pure%22%20and%20therefore%20safe%20to%20prune%20if%20unused.). By adding `﻿"sideEffects": false` to the internal package's ﻿`package.json` Next.js was able to successfully tree-shake the package.

I'm starting this discussion because I'm not an expert on library bundling, and I'm unsure if what I'm doing is incorrect in any way. However, I thought it would be a valuable addition to the [internal package's documentation](https://turbo.build/repo/docs/handbook/sharing-code/internal-packages) if it is indeed important.

**Comment by anthonyshew** [comment]
Thank you for this mention. I've made an issue out of this to get this taken care of: #6513
